src/DynamicData/Cache/IKeyValueCollection.cs

```diff
@@ -26,7 +26,7 @@ public interface IKeyValueCollection<TObject, TKey> : IEnumerable<KeyValuePair<T
     /// <value>
     /// The count.
     /// </value>
-    int Count { get; }
+    new int Count { get; }
```

Comment:
Is this an attempt to avoid a breaking change? Should we not just remove these two properties entirely? Or get rid of `IKeyValueCollection<TObject, TKey>` entirely, and replace it with `IReadOnlyList<KeyValuePair<TKey, TObject>>` everywhere it's used?

Reply:
I don't think we want to mask the base members.  I think we can just define it like this and be good:

```csharp
public interface IKeyValueCollection<TObject, TKey> : IReadOnlyList<KeyValuePair<TKey, TObject>>
{
}
```

But it is a *little* confusing having it be called `Collection` when it really is a `List`.  `Collection` implies no indexer, which it absolutely has, but it's too late to fix that, I think.

Reply:
> Is this an attempt to avoid a breaking change? Should we not just remove these two properties entirely? Or get rid of `IKeyValueCollection<TObject, TKey>` entirely, and replace it with `IReadOnlyList<KeyValuePair<TKey, TObject>>` everywhere it's used?

I don't think we can get rid of `IKeyValueCollection` because it would be a breaking change, but should be derived from a List interface instead of IEnumerable because it adds Count and an indexer.

Reply:
Yeah, I missed the fact that there were members on the interface, other than just these two, cause github collapses most of the file.

I did a little more research into it as well, and yes, this is a binary-breaking change, regardless of whether we shadow the members or not, so there's really no point in keeping them.